### ISSUE TITLE
padding input sequence changes the model's prediction

### ISSUE THREAD

**Issue by zmf0507** [QData/TextAttack#413]
while using text-attack pre-trained LSTM models like lstm-sst2, for some sentences , the model gives different predictions when input is padded and when its not. Ideally padding should not affect the model output. Can anyone give some insight and help me figure this out?
I am evaluating the model for single sentence (batch-size=1) 
I pass my sentence string to tokenizer.encode() to get input_ids which are then passed to model.forward(). The input_ids are returned with padding token at the end till max length (128). However, if I remove padding token id from input_ids and pass  it to model.forward() , the predictions come out to be different from what came earlier.

**Comment by qiyanjun** [comment]
@tahmid-kazi can you take a look at this?

**Comment by lethaiq** [comment]
@zmf0507 from my experience when working with textattack, if I use textattack's pretrained model, I should use its provided tokenizer as well, so make it consistency. If I use my own external model, I will use my own tokenizer. In other words, if I train with padding tokens, I should test with padding tokens and vice versa.

**Comment by zmf0507** [comment]
@lethaiq This makes sense but the sole purpose of padding tokens is to make variable-length sequences the same size in a batch and padding-tokens are not processed while training. That is, model learns to ignore them. Therefore, padding and no-padding should not have any differences in results. Am I right ? 
This difference is just pointing towards some mistake in training the models. That's why i raised the issue.

**Comment by qiyanjun** [comment]
@lethaiq yes. That was our design
